docs/ops/state/large_state_tuning.md

```diff
@@ -210,6 +211,71 @@ and not from the JVM. Any memory you assign to RocksDB will have to be accounted
+  User could also configure `taskmanager.memory.task.off-heap.size` to set additional quota in off-heap memory, which should be equal to `taskmanager.numberOfTaskSlots` * ``state.backend.rocksdb.memory.fixed-per-slot``, to fit in Flink's memory model.
+
+Flink also provides two parameters to tune the memory fraction of MemTable and index & filters:
+  - `state.backend.rocksdb.memory.write-buffer-ratio`, by default `0.5`. If RocksDB memory bounded feature is turned on, 50% of memory size would be used by write buffer manager by default.
+  - `state.backend.rocksdb.memory.high-prio-pool-ratio`, by default `0.1`.
+  If RocksDB memory bounded feature is turned on, 10% 0f memory size would be set as high priority for index and filters in shared block cache by default.
+  By enabling this, index and filters would not need to compete against data blocks for staying in cache to minimize performance problem if those index and filters are evicted by data blocks frequently.
+  Moreover, the L0 level filter and index are pinned into the cache by default to mitigate performance problems,
+  more details could refer to the [RocksDB-documentation](https://github.com/facebook/rocksdb/wiki/Block-Cache#caching-index-filter-and-compression-dictionary-blocks).
+
+<span class="label label-info">Note</span> The shared `cahe` and `write buffer manager` will override customized settings of block cache and write buffer via `PredefinedOptions` and `OptionsFactory`.
+
+#### Tune performance when bounding RocksDB memory usage.
+
+There might existed performance regression compared with previous no-memory-limit case if you have too many states per slot.
+If you observed this behavior and not running jobs in containerized environment or does not care about the over-limit memory usage.
+The easiest way to wipe out the performance regression is to disable memory bound for RocksDB, e.g. turn `state.backend.rocksdb.memory.managed` as false.
+Otherwise you need to increase the upper memory for RocksDB:
+  - Increase the managed memory size of `taskmanager.memory.managed.size` or the fraction via `taskmanager.memory.managed.fraction`.
+  - Increase the memory size of `state.backend.rocksdb.memory.fixed-per-slot`, which is not integrated with managed memory of task manager, as well as increasing `taskmanager.memory.task.off-heap.size` accordingly.
+
+Apart from increasing total memory, you could also tune some options:
+  - Decrease the arena block size, which is 1/8 of write-buffer size by default. This targets to decrease the possibility of turning mutable mem-table as immutable when write buffer manager reserve memory at the granularity of arena block.
+  Increase the max background flush threads for each DB instance. This targets to flush immutable mem-tables as fast as possible to not block or stall writes.
+  User could use options factory below to tune:
```

Comment:
I'm not sure whether we need to emphasize these two options (especially for the "bounding RocksDB memory usage" case), for two reasons:
1. For non-expert users, it's hard for them to understand and may cause confusion. Too many suggestions means no suggestion for non-experts, IMHO.
2. For expert users who know well about RocksDB, these are not something new. They may already tune these (and even more) options in their environment.

Reply:
I prefer to leave them here since this documentation is already be named as "tuning".

Reply:
The information given here are all RocksDB internals. As a user, I could hardly tell in what case I should decrease the arena block size or increase the background flush threads.

What's more, why do these two configurations value most? Why not tune other configurations? Are we giving the suggestion based on our own experimental case? Does it fit for all cases?

I insist that giving recommendation on tuning Flink settings instead of RocksDB ones would help more.